python/ray/tune/schedulers/hb_bohb.py

```diff
@@ -97,10 +97,20 @@ def on_trial_result(
         if not bracket.filled() or any(
             status != Trial.PAUSED for t, status in statuses if t is not trial
         ):
+            # BOHB Specific. This hack existed in old Ray versions
+            # and was removed, but it needs to be brought back
+            # as otherwise the BOHB doesn't behave as intended.
+            # There should be a better API for this.
+            # TODO(team-ml): Refactor alongside HyperBandForBOHB
```

Comment:
Instead of saying that the next line is required, can you explain why it is? Is the regular trial runner handling too late (the information has to be available in process_bracket already?)

Reply:
I clarified is as much as I could